hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/multitenant/TestMultiTenantVolume.java

```diff
@@ -77,14 +89,81 @@ public void testDefaultS3Volume() throws Exception {
     assertS3BucketNotFound(store, bucketName);
   }
 
+  private void expectFailurePreFinalization(VoidCallable eval)
+      throws Exception {
+    LambdaTestUtils.intercept(OMException.class,
+        "cannot be invoked before finalization", eval);
+  }
+
+  /**
+   * Perform sanity checks before triggering upgrade finalization.
+   */
+  private void preFinalizationCheck(ObjectStore store, String tenant,
```

Comment:
I missed this the first time looking through, but since we are sharing a cluster to save time, we should make sure the upgrade happens before either of the two tests, because we want `testDefaultS3Volume` to run on the finalized version as well. Maybe we can make the upgrade steps their own test, and define an order for the tests. Since we are using junit4, I think this would have to be done with `@FixMethodOrder(MethodSorters.NAME_ASCENDING)`, and adding a prefix to the test names to indicate their order.

Reply:
Or we could run the upgrade steps as part of the cluster setup, and if they fail it will show as a setup failure.

Reply:
Good point.

I like the idea of moving the pre-finalization checks to the cluster setup stage. Will do.